core/src/main/scala/org/apache/spark/deploy/client/StandaloneAppClient.scala

```diff
@@ -321,4 +321,15 @@ private[spark] class StandaloneAppClient(
     }
   }
 
+  /**
+   * Notify the Master about the removal of the executor
+   */
+  def executorRemoved(executorId: String): Unit = {
+    if (endpoint.get != null && appId.get != null) {
+      endpoint.get.send(ExecutorRemoved(appId.get, executorId))
+    } else {
+      logWarning("Attempted to notify removed executor before driver fully initialized.")
```

Comment:
could we also include the executorId and appId in the warning message?

Reply:
We can add `executorId` but maybe not `appId` since it's not always initialized at this time.